omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PersonResource1_8.java

```diff
@@ -54,6 +55,8 @@
 //order must be greater than that for PatientResource(order=0) RESTWS-273
 public class PersonResource1_8 extends DataDelegatingCrudResource<Person> {
 	
+	private Person person = null;
```

Comment:
Any inevitable reason of making this a class level variable?

Reply:
I need a way to reference the `person` in question  and in this context can only be initialized by the nearest defined method `getByUniqueId(String uuid)` in this class as I have refactored. @samuelmale , could there be an alternative to exposing this `person` object to this resource class without refactoring any super class to satisfy this need?

Reply:
I believe the original `getByUniqueId` method was enough to do what you want.
What happens if you remove this from the class level and edit yr changes. Does the test fail?

Reply:
For me to call this method within this resource class, I need to supply the `uuid` argument which is not anywhere referenced atleast in current class context. This, on the other hand is handled for me or called by DelegatingCrudResource-supper class [here](https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/DelegatingCrudResource.java#L51) and so with the changes I have made, `person` object is initialized at Run time when the method is called in the line I have referred to.

Reply:
Makes sense to me.

Reply:
These resources should be stateless. So can we remove the class level variable?

Reply:
Enforcing this constraint for this particular use-case implies making a huge refactor on the `getRepresentationDescription(Representation rep)` method to `getRepresentationDescription(Representation rep Person person)` or it's more generic form where all the implementing classes have to be refactored as well, which is not work related to this ticket being addressed. @dkayiwa , could there be a better way of passing reference, `delegate`/`person` to this resource without affecting the rest of the resources, or I create another ticket to refactor `getRepresentationDescription(Representation rep)` for all implementing resources?

Reply:
We are gonna have to look for another solution because resources should be stateless.